packages/dev/core/src/FlowGraph/Blocks/Data/flowGraphConditionalDataBlock.ts

```diff
@@ -0,0 +1,44 @@
+import type { FlowGraph } from "../../flowGraph";
+import { FlowGraphBlock } from "../../flowGraphBlock";
+import type { FlowGraphDataConnection } from "../../flowGraphDataConnection";
+
+/**
+ * @experimental
+ */
+export interface IFlowGraphConditionalDataBlockParams<T> {
+    defaultTrueValue: T;
+    defaultFalseValue: T;

```

Comment:
are these meant to be required? I guess it is hard to know a "default" value of a type you don't know. Maybe we could create a general "get default value"  function that will be type-dependent, and return this if those are not provided (and make them optional)?

Reply:
@RaananW Hmmm I had thought about it and it would be good to not have to specify default values especially for output nodes, but what happens if the user passes a value that is not covered on the "getDefaultValue" function? Unless we constrain the types so that the user can only pass a type that is covered, or a type that contains its own "getDefaultValue".

Reply:
we can have an adaptor that test the output type and returns a default value(0, vector2.Zero, etc..)